src/databricks/labs/ucx/workspace_access/scim.py

```diff
@@ -108,3 +108,24 @@ def _safe_get_group(self, group_id: str) -> Group | None:
         except NotFound:
             logger.warning(f"removed on backend: {group_id}")
             return None
+
+    @retried(on=[InternalError])
```

Comment:
we can't afford unbounded retries. add a timeout, either in annotation or in configuration - see example in https://github.com/databrickslabs/ucx/pull/659/files